src/model.js

```diff
@@ -104,6 +122,424 @@ CKEDITOR.define( [ 'emittermixin', 'ckeditorerror', 'utils' ], ( EmitterMixin, C
+	 * Updates the model attribute with given value. If an attribute does not exist,
+	 * it is created on the fly.
+	 *
+	 * @private
+	 * @param {Model} model The model which attribute is updated.
+	 * @param {String} attrName The name of the attribute.
+	 * @param {*} value The value of the attribute.
+	 */
+	function updateModelAttr( model, attrName, value ) {
+		if ( model.attrName ) {
+			model[ attrName ] = value;
+		} else {
+			model.set( attrName, value );
+		}
+	}
+
+	/**
+	 * Updates all bound attributes of `chain._bindModel` with the `value` of
+	 * `attrName` of `chain._current` model.
+	 *
+	 *		// Given that A == _bindModel and B == _current.model
+	 *		A.bind( 'a', 'b', 'c' ).to( B, 'x', 'y', 'x' );
+	 *
+	 *		// The following is updated
+	 *		A.a = A.c = B.x;
+	 *		A.b = B.y;
+	 *
+	 * @private
+	 * @param {BindChain} chain The chain initialized by {@link Model#bind}.
+	 * @param {String} attrName One of the attributes of `chain._current`.
+	 * @param {*} value The value of the attribute.
+	 */
+	function updateModelAttrs( chain, attrName, value ) {
+		const boundAttrs = getModelBindingsToCurrent( chain )[ attrName ];
+
+		if ( !boundAttrs ) {
+			return;
+		} else if ( chain._callback ) {
+			// MODEL.bind( 'a' ).to( TOMODEL1, 'b1' )[ .to( TOMODELn, 'bn' ) ].as( callback )
+			//  \-> Collect specific attribute value in the boundTo.model (TOMODELn.bn).
+			//
+			// MODEL.bind( 'a' ).to( TOMODEL1 )[ .to( TOMODELn ) ].as( callback )
+			//  \-> Use model attribute name to collect boundTo attribute value (TOMODELn.a).
+			const values = chain._boundTo.map( boundTo => {
+				return boundTo.model[ boundTo.attrs.length ? boundTo.attrs[ 0 ] : chain.attrs[ 0 ] ];
+			} );
+
+			// Pass collected attribute values to the callback function.
+			// Whatever is returned it becomes the value of the model's attribute.
+			updateModelAttr(
+				chain._bindModel,
+				chain._bindAttrs[ 0 ],
+				chain._callback.apply( chain._bindModel, values )
+			);
+		} else {
+			// MODEL.bind( 'a' ).to( TOMODEL1 )[ .to( TOMODELn ) ];
+			//  \-> If multiple .to() models but **no** .as( callback ), then the binding is invalid.
+			if ( !chain._callback && chain._boundTo.length > 1 ) {
+				value = undefined;
+			}
+
+			for ( let boundAttrName of boundAttrs ) {
+				updateModelAttr( chain._bindModel, boundAttrName, value );
+			}
+		}
+	}
+
+	/**
+	 * Starts listening to changes in `chain._current.model` to update `chain._bindModel`
+	 * attributes. Also sets the initial state of `chain._bindModel` bound attributes.
+	 *
+	 * @private
+	 * @param {BindChain} chain The chain initialized by {@link Model#bind}.
+	 */
+	function setupBinding( chain ) {
+		// If there's already a binding between the models (`chain._bindModel` listens to
+		// `chain._current.model`), there's no need to create another `change` event listener.
+		if ( !getModelBindingsToCurrent( chain ) ) {
+			chain._bindModel.listenTo( chain._current.model, 'change', ( evt, attrName, value ) => {
```

Comment:
I commented this line directly on a commit. For the reference: https://github.com/ckeditor/ckeditor5-core/pull/94#commitcomment-14682505